aip/general/0213.md

```diff
@@ -18,24 +18,105 @@ into taking major version changes or even lead to dependency conflicts.
 
 However, there are also cases where common structures are valuable, especially
 where a concept is well-known and it is sufficiently clear that it will not
-change. Common protos serve this use case.
+change. Common protos serve this use case. There are two forms of common protos:
+organization-specific and global.
 
-## Guidance
+## Guidance for API protos
 
-The public representation of APIs **should** be self-contained, meaning that
+The public representation of APIs **must** be self-contained, meaning that
 all protos used by the API originate in the same proto `package`, except for
 common protos, which **may** be used freely in any API.
 
 APIs **must not** define a set of API-specific common protos which live outside
-of its versioning structure. This prevents independent movement of particular
-versions and also causes problems for client libraries in many languages that
-compile the proto messages into classes.
+their versioning structure. (This is sometimes attempted as a way of reusing
+messages across multiple versions. The complexities introduced are definitely
+not worth the benefits.)
 
-APIs **should not** directly depend on protos defined in other APIs. Instead,
+APIs **must not** directly depend on protos defined in other APIs. Instead,
 they should copy and paste the applicable messages into their own API. When
-doing so, APIs **should** keep the field names and numbers the same.
-
-## Existing common protos
+doing so, APIs **should** keep the field names and numbers the same. To refer
+to resources from other APIs, specify [resource names](https://aip.dev/122)
+using `string` fields.
+
+### Rationale
+
+When one API depends on protos defined by another API, this introduces
+uncertainty in terms of customer-expected behavior and client library
+dependency management. Suppose `google.cloud.library.v1` depends on
+the protos (rather than abstract resources) in `google.cloud.movies.v2`. Any
+change to `google.cloud.movies.v2` can cause problems. For example:
+
+- If a field is added to a message in `google.cloud.movies.v2`, should
+  customers using `google.cloud.library.v1` expect to see it? If so, how soon
+  after the field has been added? What about other API changes?
+- If the whole major version is deprecated (typically after v3 has been
+  released), does that mean `google.cloud.library.v1` has to change to use
+  `google.cloud.movies.v3`, and if so, does that require a new major version for
+  the library API as well?
+- How should client library versioning reflect changes to dependent APIs?
+
+Keeping APIs isolated from each other, with a limited set of common protos which
+are maintained in a highly disciplined way, reduces a lot of the issues with
+dependencies.
+
+## Organization-specific common protos
```

Comment:
Can you put a bulleted list at the top? something like:

```
- organizations *may* have their own common protos.
- organizations *should* only have protos that are specific to their organization. Otherwise, the proto *should* live in a global common proto package.
```

Reply:
Happy to do a bulleted list to start with, but I'd then prefer the detailed paragraphs not to be bulleted. The intention that they're not just skimmed as single-sentence rules, but understood in terms of broader reasoning.

*Maybe* this means a lot of the material isn't actually suitable for AIPs though. We definitely want it documented somewhere, but this is a slightly more fuzzy topic than a lot of the things we cover in AIPs. I suspect this will now take most of our 1:1 time...